mm2src/lp_ordermatch.rs

```diff
@@ -5020,6 +5070,10 @@ pub async fn orders_kick_start(ctx: &MmArc) -> Result<HashSet<String>, String> {
     let saved_taker_orders = try_s!(storage.load_active_taker_orders().await);
 
     for order in saved_maker_orders {
+        let ctx_weak = ctx.weak();
+        let ticker = order.base.clone();
+        spawn(async move { check_balance_update_loop(ctx_weak, ticker).await });
```

Comment:
It seems that it can be spawned more than one time. E.g. if order is canceled, but new one is created fast enough (between loop iterations), we will have more and more background loops.

Reply:
I don't understand how this happens. `saved_maker_orders` is taken only once from the storage, so any cancelled/created maker_order wouldn't have any impact on the loop. Or am I missing something?

Reply:
But yes this happens within the 30 second intervals in the `check_balance_update_loop`. Multiple balance loops for the same coin can be spawned by continuously cancelling/creating orders. I'll look for a solution.